### PR TITLE
Remove deprecated print function in ConsoleReport

### PR DESCRIPTION
**Pull request by marschwar**
This removes the deprecated `print` function from the `ConsoleReport` class

### CODE REVIEW

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ConsoleReport.kt [hunk 1]
```diff
@@ -11,18 +9,6 @@ import java.io.PrintStream
  */
 abstract class ConsoleReport : Extension {
```
Comment (marschwar) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ConsoleReport.kt:10]: Should we change this to an interface?

Comment (schalkms) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ConsoleReport.kt:10]: Can we remove this abstract class completely?

**Review by schalkms — APPROVED** [review]

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ConsoleReport.kt (hunk 1 above)
Comment (marschwar) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ConsoleReport.kt:10]: I don't think so. This is one of the  [public extension points to create custom reports](https://detekt.dev/docs/introduction/extensions#custom-reports).

Comment (schalkms) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ConsoleReport.kt:10]: Yeah. We have the same `render()` function in both reports now.
I don't think the design for the reports fits that well now.

**Pull request merged**